### PR TITLE
server: remove username required condition for listVmsForImport

### LINKED ISSUE #8414: Minor: listVmsForImport API fix
ACS 4.19 RC1
- minor update

listVmsForImport API states (based on the docs) that the username and password parameters are required (besides zoneid, host IP and hypervisor=KVM), while in reality, this API seems to connect to remote libvirt via TCP unauthenticated connections and lists stopped VMs.

Using CMK if the "username" parameter is omitted - it gives error and says it's mandatory, 

```
(localcloud)  > listVmsForImport zoneid=fbfd1039-ca49-436d-b17d-e03f39663047 hypervisor=KVM host=10.0.35.106
 Error: (HTTP 431, error code 4350) Username need to be provided.
```

...however any random username can be passed and it will work:

```
(localcloud)  > listVmsForImport username=aaaaaaaaaaaaaaaaa zoneid=fbfd1039-ca49-436d-b17d-e03f39663047 hypervisor=KVM host=10.0.35.106 
{
  "count": 1,
  "unmanagedinstance": [
    {
      "cpucorepersocket": -1,
      "disk": [
        {
          "capacity": 17179869184,
          "controller": "virtio",
          "datastorepath": "/var/lib/libvirt/images/MasteringKVM01.qcow2",
          "datastoretype": "file",
          "id": "0",
          "label": "vda",
          "position": 0
        },
        {
```

If fixed, the docs also needs to be updated to remove the statement that the username/password are mandatory

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #8414

Doc changes added https://github.com/apache/cloudstack-documentation/pull/401

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by vishesh92 — APPROVED** [review]
clgtm. didn't test.

**Comment by DaanHoogland** [comment]
@shwstppr as the parameters are not used, should these also be removed from the `Cmd` class?

**Comment by weizhouapache** [comment]
> @shwstppr as the parameters are not used, should these also be removed from the `Cmd` class?

currently vms are listed by virsh command (endpoint is the remote server)
However, it does not work in some cases (e.g. libvirt uses ssl).
username/password could be useful in the future

**Comment by vishesh92** [comment]
@shwstppr Do we need any changes in the UI for this?

**Comment by andrijapanicsb** [comment]
@vishesh92 no - those are used later during importVm API to connect via ssh etc.

**Review by rohityadavcloud — APPROVED** [review]

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by vladimirpetrov — APPROVED** [review]
LGTM based on manual testing - the username parameter is no longer mandatory.

File: server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java [hunk 1]
```diff
@@ -2042,10 +2042,6 @@ private UserVmResponse importKvmInstance(ImportVmCmd cmd) {
 
         UnmanagedInstanceTO unmanagedInstanceTO = null;
         if (ImportSource.EXTERNAL == importSource) {
-            if (StringUtils.isBlank(cmd.getUsername())) {
-                throw new InvalidParameterValueException("Username need to be provided.");
-            }
-
```
[the remark this answers is not in the document]
Comment (shwstppr) [reply server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java]: done

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
Merged based on GHA smoketests and build tests.

**Pull request merged**